mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -507,6 +510,10 @@ LogicalResult Conv2dGenerator::genConvModule(ModuleOp &module, int kernel_id,
     return failure();
   }
 
+  Type outputDataType = dataType;
+  if (dataType.isInteger(8)) {
+    outputDataType = builder.getIntegerType(32);
+  }
```

Comment:
Irrelevant to where I put the review...

I'd prefer a small end-to-end test (one of all 1, another of all 0s, for example) that exercise i8 convolution and get expected result by just using `miopen-gen`.